### PR TITLE
Rem 514

### PR DESCRIPTION
**Pull request by Ghinwa-Arab**
Assign Outwards Correspondent
it's table that has checkbox column, data filters according to grid toolbar filters (country mandatory and the rest optional) where it should return outwards transfer data based on selected filters && has no correspondent.
then after selecting outwards we can open correspondent popup placed below the table and assign correspondent for them

### CODE REVIEW

File: src/pages/assign-out-corr/index.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+const OutwardsCorrespondent = () => {
+  const { getRequest } = useContext(RequestsContext)
+  const { stack } = useWindow()
+  const [data, setData] = useState([])
+
+  const {
+    labels: labels,
+    maxAccess,
+    refetch
```
Comment (omarZrayka23) [inline src/pages/assign-out-corr/index.js:28]: u can use invalidate instead refetch

File: src/pages/assign-out-corr/AssignCorrespondentForm.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import { RemittanceSettingsRepository } from 'src/repositories/RemittanceRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { RemittanceOutwardsRepository } from 'src/repositories/RemittanceOutwardsRepository'
+import { useForm } from 'src/hooks/form'
+
+export default function AssignCorrespondentForm({ maxAccess, labels, outwardsList, refetch, window }) {
+  const { postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
```
Comment (omarZrayka23) [inline src/pages/assign-out-corr/AssignCorrespondentForm.js:17]: u can use Invalidate instead refetch props

File: src/pages/assign-out-corr/index.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    stack({
+      Component: AssignCorrespondentForm,
+      props: {
+        maxAccess: maxAccess,
+        labels: labels,
+        outwardsList: data?.list?.filter(item => item.checked),
+        refetch: fetchWithFilter
+      },
+      width: 600,
+      height: 500,
```
Comment (omarZrayka23) [inline src/pages/assign-out-corr/index.js:97]: decrease height 
Screenshot 2024-08-22 at 11 09 06 AM

File: src/pages/assign-out-corr/index.js [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                  { key: 'name', value: 'Name' },
+                  { key: 'flName', value: 'Foreign Language Name' }
+                ]}
+                values={formik.values}
+                displayFieldWidth={1.75}
+                onChange={(event, newValue) => {
+                  if (newValue) {
+                    formik.setFieldValue('countryId', newValue?.recordId || 0)
+                  }
+                }}
```
Comment (Alinadimmokdad) [inline src/pages/assign-out-corr/index.js:152]: add else ''

**Pull request merged**